src/wp-includes/block-template-utils.php

```diff
@@ -36,22 +36,9 @@
  *     @type string $wp_template_part Theme-relative directory name for block template parts.
  * }
  */
-function get_block_theme_folders( $theme_stylesheet = null ) {
-	$theme_name = null === $theme_stylesheet ? get_stylesheet() : $theme_stylesheet;
-	$root_dir   = get_theme_root( $theme_name );
-	$theme_dir  = "$root_dir/$theme_name";
-
-	if ( file_exists( $theme_dir . '/block-templates' ) || file_exists( $theme_dir . '/block-template-parts' ) ) {
-		return array(
-			'wp_template'      => 'block-templates',
-			'wp_template_part' => 'block-template-parts',
-		);
-	}
-
-	return array(
-		'wp_template'      => 'templates',
-		'wp_template_part' => 'parts',
-	);
+function get_block_theme_folders( $theme_stylesheet = '' ) {
+	$theme = wp_get_theme( $theme_stylesheet );
+	return $theme->get_block_template_folders();
```

Comment:
Related to @spacedmonkey's comment in https://github.com/WordPress/wordpress-develop/pull/5202/files#r1326178073 (see also my reply):
```suggestion
	if ( ! $theme->exists() ) {
		// Return the default folders if the theme doesn't exist.
		return array(
			'wp_template'      => 'templates',
			'wp_template_part' => 'parts',
		);
	}
	return $theme->get_block_template_folders();
```

Reply:
Added this check.